### PR TITLE
Data frames: get rid of wandb-run.json files

### PR DESCRIPTION
**Pull request by adrianbg**
Start renaming data tables to data frames. Make Simpsons example update the results DataFrame after each epoch as well as at the end of the run.

### CODE REVIEW

**Comment by adrianbg** [comment]
@shawnlewis 
This changes summary semantics: before, large structures like tensors would be written to files and replaced in the summary with a placeholder dictionary. This meant that if the user looked for their tensor after writing it to the summary, they would find the placeholder dictionary instead. If they then put the same tensor object into the summary again, except with different data in it, it would be treated as a new tensor and serialized all over again. The placeholder dictionary was effectively acting as a "clean/dirty" marker.

Now we track such objects by their Python ID's (ie. their memory addresses). The placeholder dictionaries don't magically appear in the summary any more, so we have no way of noticing if a particular summary key is dirty unless they put an entirely new tensor object in the same spot.

**Comment by adrianbg** [comment]
@shawnlewis implemented nested summaries that notice when any key is modified, even if it's not at the top level. This fixes that nasty bug we discussed on the offsite. This replaces the solution I described in the previous comment, which you didn't like.

**Comment by adrianbg** [comment]
@shawnlewis @raubitsj could you review this?

File: wandb/summary.py [hunk 1]
```diff
@@ -14,199 +16,295 @@
...
-    def __getattr__(self, k):
-        if k.startswith("_"):
-            return super(Summary, self).__getattr__(k)
-        else:
-            return self._transform(k.strip(), self._summary[k.strip()], write=False)
+        self._locked_keys.add(k)
+
+        self._root._write()
+
+        return v
```
Comment (raubitsj) [inline wandb/summary.py]: is that return needed or doing anything?  normally it isnt set with __setitem__ definitions.

Comment (raubitsj) [inline wandb/summary.py]: is that return needed or doing anything?  normally it isnt set with __setitem__ definitions.

File: wandb/summary.py [hunk 2]
```diff
@@ -14,199 +16,295 @@
...
-        return v
+    def get(self, k, default=None):
+        k = k.strip()
+        if k not in self._dict:
+            self._root._root_get(self._path + (k,), self._dict)
+        return self._dict.get(k, default)
 
     def __getitem__(self, k):
-        return self._transform(k, self._summary[k], write=False)
+        self.get(k)  # load the value into _dict if it should be there
```
Comment (raubitsj) [inline wandb/summary.py]: should there be a: ```k=k.strip() before this line?```

File: wandb/summary.py (hunk 1 above)
Comment (adrianbg) [reply wandb/summary.py]: I'd take your word for that. I assumed it was needed for eg. `a = b['key'] = c` but I don't think I actually checked.

File: wandb/summary.py (hunk 2 above)
Comment (adrianbg) [reply wandb/summary.py]: Yep. Thanks.

**Pull request merged**